pkg/validator/validators_test.go

```diff
@@ -10,13 +9,204 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+func TestIngressBackendValidator(t *testing.T) {
+	assert := tassert.New(t)
+	testCases := []struct {
+		name      string
+		input     *admissionv1.AdmissionRequest
+		expResp   *admissionv1.AdmissionResponse
+		expErrStr string
+	}{
+		{
+			name: "IngressBackend with valid protocol succeeds",
+			input: &admissionv1.AdmissionRequest{
+				Kind: metav1.GroupVersionKind{
+					Group:   "v1alpha1",
+					Version: "policy.openservicemesh.io",
+					Kind:    "IngressBackend",
+				},
+				Object: runtime.RawExtension{
+					Raw: []byte(`
+					{
+						"apiVersion": "v1alpha1",
+						"kind": "IngressBackend",
+						"spec": {
+							"backends": [
+								{
+									"name": "test",
+									"port": {
+										"number": 80,
+										"protocol": "http"
+									}
+								}
+							]
+						}
+					}
+					`),
+				},
+			},
+			expResp:   nil,
+			expErrStr: "",
+		},
+		{
+			name: "IngressBackend with invalid protocol errors",
+			input: &admissionv1.AdmissionRequest{
+				Kind: metav1.GroupVersionKind{
+					Group:   "v1alpha1",
+					Version: "policy.openservicemesh.io",
+					Kind:    "IngressBackend",
+				},
+				Object: runtime.RawExtension{
+					Raw: []byte(`
+					{
+						"apiVersion": "v1alpha1",
+						"kind": "IngressBackend",
+						"spec": {
+							"backends": [
+								{
+									"name": "test",
+									"port": {
+										"number": 80,
+										"protocol": "invalid"
+									}
+								}
+							]
+						}
+					}
+					`),
+				},
+			},
+			expResp:   nil,
+			expErrStr: "Expected 'Port.Protocol' to be 'http|https', got: invalid",
```

Comment:
Might be overkill for now, but one way to get around hardcoding error strings in tests like this is to define a new type that implements the `error` interface or a function that returns an error so then in the code you can `return badPortProtocol{protocol: "invalid"}` or `return badPortProtocol("invalid")` and use the same type/function here in the tests. That makes tweaking the messages easy without having to worry about the tests.

Reply:
I am not a big fan of hardcoding error messages either, but I didn't want to refactor the existing tests too much, so the scope of this PR remains as small as possible. This can be an enhancement that can be done later on.